libsolutil/CommonData.h

```diff
@@ -404,7 +404,8 @@ inline std::string toHex(u256 val, HexPrefix prefix = HexPrefix::DontAdd)
 	return (prefix == HexPrefix::Add) ? "0x" + str : str;
 }
 
-inline std::string toCompactHexWithPrefix(u256 const& _value)
+template <class T>
+inline std::string toCompactHexWithPrefix(T _value)
```

Comment:
`toCompactBigEndian` is a similar template allowing for any unsigned integer and `bigint`-convertible type.

Reply:
@axic if it's only two different types that are allowed for the template type `T` and the body is basically a one-liner, wouldn't it make more sense to simply provide two overloads? It would reduce that question of what `T` can all be IMHO.

Reply:
The typecheck is enforced in `toCompactBigEndian`. If you think it helps, that could be changed to having two overloads, but not sure that fits this PR well?

Reply:
I'm fine.